doc/musig.md

```diff
@@ -0,0 +1,51 @@
+Notes on the musig module API
+===========================
+
+The following sections contain additional notes on the API of the musig module (`include/secp256k1_musig.h`).
+A usage example can be found in `examples/musig.c`.
+
+# API misuse
```

Comment:
nit:

```suggestion
## API misuse
```

(also for all other headings below)

`#` is the same level as the heading for the entire document, see the rendered file.

Reply:
fixed